src/mem/largealloc.h

```diff
@@ -153,8 +153,10 @@ namespace snmalloc
       // Register this allocator for low-memory call-backs
       if constexpr (pal_supports<LowMemoryNotification, PAL>)
       {
-        allocated->PalNotificationObject::pal_notify = &(allocated->process);
-        PAL::register_for_low_memory_callback(allocated);
+        auto callback =
+          allocated->template alloc_chunk<LowMemoryNotificationObject, 1>(
+            allocated);
+        PAL::register_for_low_memory_callback(callback);

```

Comment:
Does this need to be a dynamically allocated thing?  Can we make it a field and use the `std::conditional` and no_unique_address trick to make it not use any space if not used?

Reply:
No unique address is a C++20 thing.  @nwf found issues when we tried to move to C++20 with the instantiation of the pagemap causing compile times to be insanely large.

Reply:
The main issue is it is a little complex for the construction to refer to the enclosing type by pointer.  So it would be much less nice code when embedded.

Reply:
I could manually combine the allocation a few lines higher:
https://github.com/microsoft/snmalloc/blob/ee470c535e258d93590488a4e1266bbe24dc90f1/src/mem/largealloc.h#L142-L145

This could modify the size to account for other allocations. However, variable allocation sizes like that seem to regularly be related to security vulnerabilities.  So I am not keen to introduce the anti-pattern.

Reply:
Okay, I have modified the address_space to allow for allocations below a power of two, and the address space will retain the extra.  Thus is you ask for 
  1. 1500 bytes
  2. 16 bytes
  3. 16 bytes 

Then it will place the 16 bytes in the gap between 1500 - 2048. This makes it cheaper to allocate multiple objects for clarity, while not incurring the rounding costs for the allocations.

Reply:
@davidchisnall thoughts?  What are you reasons for not wanting an allocation?